src/math/Bspline.cpp

```diff
@@ -0,0 +1,75 @@
+#include "math/Bspline.hpp"
+#include "math/differences.hpp"
+#include "utils/assertion.hpp"
+
+#include <Eigen/Core>
+#include <algorithm>
+#include <cstdlib>
+#include <unsupported/Eigen/Splines>
+#include "math/differences.hpp"
+#include "utils/assertion.hpp"
+
+namespace precice::math {
+
+Bspline::Bspline(Eigen::VectorXd ts, const Eigen::MatrixXd &xs, int splineDegree)
+{
+
+  PRECICE_ASSERT(ts.size() >= 2, "Interpolation requires at least 2 samples");
+#if EIGEN_VERSION_AT_LEAST(3, 4, 0)
+  PRECICE_ASSERT(std::is_sorted(ts.begin(), ts.end()), "Timestamps must be sorted");
+#else
+  PRECICE_ASSERT(std::is_sorted(ts.data(), ts.data() + ts.size()), "Timestamps must be sorted");
+#endif
+
+  // organize data in columns. Each column represents one sample in time.
+  PRECICE_ASSERT(xs.cols() == ts.size());
+  _ndofs            = xs.rows(); // number of dofs. Each dof needs its own interpolant.
+  _tsMin            = ts(0);
+  _tsMax            = ts(ts.size() - 1);
+  auto relativeTime = [this](double t) -> double { return (t - _tsMin) / (_tsMax - _tsMin); };
```

Comment:
I honestly don't know if this is well-defined, as you capture `this` by value inside the constructor. I would go safe instead and simplify this by capturing `_tsMin` and `_tsMax` by value instead.

```suggestion
  auto relativeTime = [_tsMin, _tsMax](double t) -> double { return (t - _tsMin) / (_tsMax - _tsMin); };
```